crates/ruff_linter/src/checkers/ast/analyze/expression.rs

```diff
@@ -1517,7 +1517,7 @@ pub(crate) fn expression(expr: &Expr, checker: &mut Checker) {
                 ruff::rules::parenthesize_chained_logical_operators(checker, bool_op);
             }
         }
-        Expr::NamedExpr(..) => {
+        Expr::Named(..) => {
```

Comment:
This one specifically _does_ feel slightly less clear to me (not sure why)...

As an alternative, although the "walrus operator" has strictlyy-speaking only ever been a nickname for this piece of syntax -- we _could_ consider calling this variant `Expr::Walrus`. It would be immediately clear to the reader what this variant is referring to!

Reply:
I don't mind `Walrus` and might prefer it over `NamedExpr` but it diverges from the Python AST naming (wouldn't be the first time, we renamed `FormattedString` to `FString`). The only issue it has is that it is not very approachable for users new to Python: I remember last year at PyCon when someone started talking to me about walrus operators and I was completely lost. 

I would prefer to keep any potential rename to its own PR. It might deserve a dedicated discussion

Reply:
> I remember last year at PyCon when someone started talking to me about walrus operators and I was completely lost.

Ah, but _I_ see that as a point in _favour_ of renaming it to `Walrus` ;) I think approximately 0 Python users refer to `:=` as the "named expression operator", or the "assignment expression operator". And just about every mention in CPython's docs refers to it as "the assignment expression operator (also known as the 'walrus operator')". If we use the more common term for it internally, maybe we'll get less confused when people try to talk to us about it at PyCon ;)

But I'm fine with postponing renaming the variant to a future PR!

Reply:
Assignment expression operator is interesting, that sounds more approachable by non python users ;) But let's leave bikeshedding out of it